src/System.Collections.NonGeneric/tests/SortedListTests.cs

```diff
@@ -211,15 +212,54 @@ public static void Ctor_IDictionary_IComparer_NullDictionary_ThrowsArgumentNullE
         }
 
         [Fact]
-        public static void DebuggerAttribute()
+        public static void DebuggerAttribute_Empty()
+        {
+            Assert.Equal("Count = 0", DebuggerAttributes.ValidateDebuggerDisplayReferences(new SortedList()));
+        }
+
+        [Fact]
+        public static void DebuggerAttribute_NormalList()
         {
             var list = new SortedList() { { "a", 1 }, { "b", 2 } };
+            DebuggerAttributeInfo debuggerAttribute = DebuggerAttributes.ValidateDebuggerTypeProxyProperties(list);
+            PropertyInfo infoProperty = debuggerAttribute.Properties.Single(property => property.Name == "Items");
+            object[] items = (object[])infoProperty.GetValue(debuggerAttribute.Instance);
 
-            DebuggerAttributes.ValidateDebuggerDisplayReferences(new SortedList());
+            Assert.Equal(list.Count, items.Length);
+            for (int i = 0; i < list.Count; i++)
+            {
+                object item = items[i];
+                object key = item.GetType().GetField("_key", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(item);
+                object value = item.GetType().GetField("_value", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(item);
```

Comment:
I don't love having reflection explicitly targeting field names like this; makes it harder to refactor.  Is this really necessary?